In [1]:
import pandas as pd
import plotly.express as px

In [2]:
df = pd.read_csv("Compute_allocations_transformer.csv")
# byte_level = df[df['Patch size'] == 1]
byte_level = df[(df['FLOPs'] == 's') & (df['Patch size'] == 2)]
# byte_level['Global transformer parameters'].sort_values(ascending=False)
fig = px.histogram(byte_level['Global model parameters'])

fig.show()

In [7]:
byte_level[(byte_level['Global model parameters'] <= 1.2e7) & (byte_level['Global model parameters'] >= 0.8e7)]

,FLOPs,Patch size,Tokens,Encoder parameters,Global model parameters,Decoder parameters,Global model dimension,Encoder/Decoder dimension,Global model layers,Decoder layers
0,s,2,6.312902e+10,442736.64,9641820.16,885473.28,448,192,4,2
54,s,2,6.392402e+10,442736.64,8854732.80,885473.28,384,192,5,2
111,s,2,5.509120e+10,442736.64,10625679.36,885473.28,384,192,6,2
173,s,2,6.081845e+10,442736.64,8608768.00,885473.28,320,192,7,2
236,s,2,5.450177e+10,442736.64,9838592.00,885473.28,320,192,8,2
294,s,2,4.937376e+10,442736.64,11068416.00,885473.28,320,192,9,2
539,s,2,4.697577e+10,442736.64,11068416.00,1328209.92,320,192,9,3
774,s,2,4.033749e+10,442736.64,11806310.40,1328209.92,256,192,15,3
887,s,2,5.678665e+10,196771.84,8657960.96,787087.36,256,128,11,4
923,s,2,5.266842e+10,196771.84,9445048.32,787087.36,256,128,12,4


4e18 FLOPs
===

tokens, global model params, decoder params, global model dimension, encoder/decoder dimension, global model layers, decoder layers

Patch size = 2

5.509120e+10, 10625679,	885473,	384, 192, 6, 2 (10M run)

1.993053e+10 31483494, 3689472, 512, 320, 10, 3 (30M run)

1.188846e+10, 59031552, 6149120, 640, 320, 12, 5 (60M run)

Patch size = 4

4.937337e+10, 21691515, 1836319, 576, 256, 7, 3

1.847138e+10, 60177695, 5508956, 704, 384, 13, 4

9.375604e+09, 120M, 12242125, 896, 512, 16, 5